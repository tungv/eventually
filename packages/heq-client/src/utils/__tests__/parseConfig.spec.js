const parseConfig = require('../parseConfig');
const hasModule = require('../has-module');
const mockServer = require('../__mocks__/mockServer');

jest.mock('../has-module');

describe('config', () => {
  let server;

  beforeAll(async () => {
    server = await mockServer(1, [{ id: 1, payload: 'test' }]);
  });

  afterAll(() => {
    server.close();
  });
  describe('config: subscribe', () => {
    it('should throw if subscribe is not defined', () => {
      return expect(parseConfig({})).rejects.toMatchSnapshot();
    });

    it('should throw if subscribe.serverUrl is not defined', () => {
      return expect(parseConfig({ subscribe: {} })).rejects.toMatchSnapshot();
    });

    it('should not throw if subscribe.burstCount or subscribe.burstTime are not defined', async () => {
      const config = await parseConfig(
        {
          subscribe: {
            serverUrl: server.url,
          },
          persist: {
            store: process.env.MONGO_TEST,
          },
          transform: {
            rulePath: '../../fixtures/rules/user_management.js',
          },
        },
        __dirname,
      );

      expect(config).toMatchObject({
        subscribe: {
          serverUrl: server.url,
          burstCount: 20,
          burstTime: 500,
        },
      });
    });

    it('should throw if subscribe.serverUrl is unreachable', () => {
      return expect(
        parseConfig({
          subscribe: {
            serverUrl: 'http://localhost:1338',
          },
        }),
      ).rejects.toMatchSnapshot();
    });
  });

  describe('config: persist', () => {
    it('should detect persist driver is defined', async () => {
      const config = {
        subscribe: { serverUrl: server.url },
        persist: {
          store: process.env.MONGO_TEST,
        },
        transform: {
          rulePath: '../../fixtures/rules/user_management.js',
        },
      };

      expect(await parseConfig(config, __dirname)).toMatchObject({
        persist: {
          store: process.env.MONGO_TEST,
          driver: '@events/snapshot-mongo',
        },
      });

      expect(hasModule).toBeCalledWith('@events/snapshot-mongo');
    });
  });
});
