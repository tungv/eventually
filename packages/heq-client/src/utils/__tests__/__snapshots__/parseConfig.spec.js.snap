// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config config: subscribe should throw if subscribe is not defined 1`] = `[Error: INVALID_CONFIG: path=\`subscribe\`]`;

exports[`config config: subscribe should throw if subscribe.serverUrl is not defined 1`] = `[Error: INVALID_CONFIG: path=\`subscribe.serverUrl\`]`;

exports[`config config: subscribe should throw if subscribe.serverUrl is unreachable 1`] = `[Error: INVALID_ENDPOINT: \`http://localhost:1338\` - reason: Error: connect ECONNREFUSED 127.0.0.1:1338]`;
